doc/src/modules/solvers/solveset.rst

```diff
@@ -346,10 +346,10 @@ How do we manipulate and return an infinite solution?
    >>> from sympy import ImageSet, Lambda, pi, S, Dummy, pprint
    >>> n = Dummy('n')
    >>> pprint(ImageSet(Lambda(n, 2*pi*n), S.Integers), use_unicode=True)
-   {2⋅n⋅π | n ∊ ℤ}
+   {2⋅_n⋅π | _n ∊ ℤ}
```

Comment:
I don't think we want to do this. If there is no ambiguity in the expression's variables, just use symbols to print the dummies. There is ambiguity when any two symbols share the same name. In such cases, the names can be given arbitrary numbers to disambiguate them. Clashes can exist in multiple ways: 
```python
>>> len(set([Dummy('x'), Dummy('x'), Symbol('x', real=True), Symbol('x', integer=True), Wild('x', exclude=[y]), Wild('x', exclude=[z])])) == 6
True
```
Bound symbols should never appear with an underscore since (I believe) their context will make them unambiguous.

Reply:
I somehow agree. It looks a bit weird. It is rather straightforward to reverse this. (At least from a code perspective, the documentation is a bit more of a hassle...) I just based it on https://github.com/sympy/sympy/issues/6498#issuecomment-461196106 so I think @asmeurer should have a say here.

Implementing adding of the underscore based on other names in the expression is far from trivial with the current `Printer` setup, since `_print_Dummy` does not have any idea about the expression.

Reply:
The user should not have to be bothered with seeing underscores if they don't add any information. So perhaps part of the printing process is to replace all Dummy/Wild that have a clashing name with Symbol *before* doing the printing. Underscores will only remain if there was a clash. (And if we don't like how it looks then we (as coder or user) need to pick variables that don't clash.
```python
def DummyWild_as_Symbol(eq):
    from sympy.core.symbol import Symbol, Dummy, Wild
    from collections import defaultdict
    free = eq.free_symbols
    bound = eq.atoms(Symbol) - free
    # replace bound symbols that are Dummy or Wild with Symbols
    # since these are never ambiguous in meaning
    eq = eq.subs(dict([(b, Symbol(b.name)) for b in bound if isinstance(b, (Dummy, Wild))]))
    # replace Dummy or Wild with Symbol if there is no ambiguity in doing so
    dups = defaultdict(list)
    for i in [i for i in free]:
        dups[i.name].append(i)
    for i in dups:
        if len(dups[i]) == 1:
            u = dups[i][0]
            if isinstance(u, (Dummy, Wild)):
                eq = eq.xreplace({u: Symbol(i)})
    return eq
```
User's who want to print `x + x` that looks nice (perhaps for pedagogical reasons) can do so with an unevaluated expression or can instantiate symbols with an unused assumption:
```
>>> x, y = [Symbol('x', _=i) for i in range(2)]
>>> x + y
x + x
```

Reply:
> user should not have to be bothered with seeing underscores

Revisiting this now, it seems that if we are seeing underscores appear then we should fix the offending code that put them there. In case of the ImageSet, a dummy should only be used if it clashes with an existing symbol.

Reply:
> user should not have to be bothered with seeing underscores

I guess that is the main question here. According to #6498 that is exactly what the users should be bothered with. I do not have any strong opinions as such. Just solved the issue. @asmeurer ?